src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -1160,4 +1174,30 @@ public void postBlUsersSettingsMyprofilePassword(UpdateCredentials entity, Map<S
         .respond500WithTextPlain("Internal server error during change user's password")));
     }
   }
+
+  @Override
+  public void postBlUsersPasswordResetValidate(Map<String, String> okapiHeaders,
+                                               Handler<AsyncResult<javax.ws.rs.core.Response>> asyncResultHandler,
+                                               Context vertxContext) {
+    OkapiConnectionParams connectionParams = new OkapiConnectionParams(okapiHeaders.get("x-okapi-url"),
+      okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT), okapiHeaders.get(RestVerticle.OKAPI_HEADER_TOKEN), "");
+
+    passwordResetLinkService.validateLinkAndLoginUser(connectionParams)
+      .setHandler(res -> {
+        if (res.succeeded()) {
+          asyncResultHandler.handle(Future.succeededFuture(
+            PostBlUsersPasswordResetValidateResponse.respond200WithApplicationJson(res.result())));
+        } else {
+          if (res.cause().getClass() == UnprocessableEntityException.class) {
+            asyncResultHandler.handle(Future.succeededFuture(
+                PostBlUsersPasswordResetValidateResponse.respond422WithTextPlain(
+                  ExceptionHelper.handleException(res.cause()))));
```

Comment:
ExceptionHelper#handleException returns Response object with status and body setup, so just pass returned object to asyncResultHandler like:
} else {
          asyncResultHandler.handle(Future.succeededFuture(ExceptionHelper.handleException(res.cause())));
        }